File: 1544-make-the-string-great/1544-make-the-string-great.js
/**
 * @param {string} s
 * @return {string}
 */
var makeGood = function(s) {
    let ans = []
    for(let x of s){
        if(ans.length === 0){
            ans.push(x)
        }else{
            if(x.toLowerCase() === ans[ans.length-1].toLowerCase()){
                if(x !== ans[ans.length -1]){
                    ans.pop()
                }else{
                ans.push(x)
                }
            }else {
                ans.push(x)

            }
        }
    }
    return ans.join("")
};

File: 1544-make-the-string-great/1544-make-the-string-great.java
public class Solution {
    public String makeGood(String s) {
        StringBuilder stack = new StringBuilder();
        
        for (char ch : s.toCharArray()) {
            int len = stack.length();
            
            if (len > 0) {
                char top = stack.charAt(len - 1);
                
                if (Character.toLowerCase(top) == Character.toLowerCase(ch)
                    && top != ch) {
                    stack.deleteCharAt(len - 1);
                    continue;
                }
            }
            
            stack.append(ch);
        }
        
        return stack.toString();
    }
}